proto/substrait/algebra.proto

```diff
@@ -255,6 +255,13 @@ message JoinRel {
   Rel left = 2;
   Rel right = 3;
   Expression expression = 4;
+  // The post-join filter is a filter that is applied to the result of the join before an output
+  // record is produced.  If the filter evaluates to false then the record is not considered a
+  // match.
```

Comment:
> ...applied to the result of the join

I believe this is semantically ambiguous for functions that aggregate over many tuples. I think you would get different results if such a function is applied over the input records (what is evaluated by `expression`) vs applied on the records passing the filter.